PetAdoptions/pethistory/index.js

```diff
@@ -0,0 +1,237 @@
+        } catch (error) {
+            console.error('Failed to process record:', record.messageId, error);
+            // Add to batch failures instead of throwing
+            batchItemFailures.push({
+                itemIdentifier: record.messageId
+            });
+        }
+    }
+
+    console.log(`Processed ${results.length} messages successfully, ${batchItemFailures.length} failed`);
+
+    // Return batch failure information for SQS to handle partial failures
+    return {
+        batchItemFailures: batchItemFailures,
+        processedCount: results.length,
+        results: results
+    };
+};
+
+/**
+ * Process a single adoption history message from SQS
+ */
+async function processAdoptionMessage(record) {
+    const messageId = record.messageId;
+    // const receiptHandle = record.receiptHandle;
+
+    console.log(`Processing message ${messageId}`);
+
+    // Parse and validate message
+    let adoptionData;
+    try {
+        adoptionData = JSON.parse(record.body);
+        console.log('Parsed adoption data:', adoptionData);
+    } catch (error) {
+        console.error('Invalid JSON in message:', messageId, error);
+        throw new Error(`Invalid JSON in message ${messageId}: ${error.message}`);
+    }
+    // Validate message schema
+    const { error, value } = adoptionMessageSchema.validate(adoptionData);
+    if (error) {
+        console.error('Schema validation failed for message:', messageId, error.details);
+        throw new Error(`Schema validation failed for message ${messageId}: ${error.message}`);
+    }
+    const validatedData = value;
+    try {
+        console.log(`Processing adoption - Transaction: ${validatedData.transactionId}, Pet: ${validatedData.petId}, User: ${validatedData.userId}`);
+        // Write adoption history to database
+        await writeAdoptionHistoryToDatabase(validatedData);
+
+        console.log(`Successfully processed adoption for pet ${validatedData.petId} by user ${validatedData.userId}`);
+
+        return {
+            messageId: messageId,
+            transactionId: validatedData.transactionId,
+            status: 'success'
+        };
+
+    } catch (error) {
+        console.error(`Failed to process adoption for pet ${validatedData.petId}:`, error);
+        throw error;
+    }
+}
+
+/**
+ * Write adoption history record to PostgreSQL database
+ */
+async function writeAdoptionHistoryToDatabase(adoptionData) {
+    let client;
+
+    try {
+        console.log(`Writing adoption history to database: ${adoptionData.transactionId}`);
+
+        // Get database connection details from Secrets Manager
+        const dbConfig = await getDatabaseConfig();
+
+        // Create database connection
+        client = new Client({
+            host: dbConfig.host,
+            port: dbConfig.port,
+            database: dbConfig.dbname,
+            user: dbConfig.username,
+            password: dbConfig.password,
+            ssl: false, // Adjust based on your RDS configuration
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It looks like your code has disabled the server certificate validation, which makes your application vulnerable to man-in-the-middle (MitM) attacks. We recommend that you set `rejectUnauthorized` to `true`.

[Learn more](https://cwe.mitre.org/data/definitions/295.html)